core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -268,11 +271,12 @@ abstract class AbstractFetcherThread(name: String,
             fetchOffsets.put(tp, offsetTruncationState)
 
         case Errors.FENCED_LEADER_EPOCH =>
-          if (onPartitionFenced(tp, latestEpochsForPartitions.get(tp).flatMap {
-            p =>
-              if (p.currentLeaderEpoch.isPresent) Some(p.currentLeaderEpoch.get())
-              else None
-          })) partitionsWithError += tp
+          val currentLeaderEpoch = latestEpochsForPartitions.get(tp) match {
```

Comment:
Maybe we can use `scala.compat.java8.OptionConverters._`?

Reply:
We could but that would require two conversions instead of one:
```
val currentLeaderEpoch = latestEpochsForPartitions.get(tp).flatMap { p =>
    RequestUtils.getLeaderEpoch(p.currentLeaderEpoch).asScala
}.asJava
```
I have a small preference for the current approach to avoid this.

Reply:
Hmm.. It's a little curious that we need the call to `getLeaderEpoch` here. Tracing this back to `fetchTruncatingPartitions`, it looks like we could not have a negative epoch here.

Reply:
Yeah, that's because `onPartitionFenced` takes an `Optional` actually.

Reply:
As onPartitionFenced is used in other code paths, I did not refactor it. We could have another variant which does not take an Optional. Is it worth doing it?

Reply:
@hachikuji Ah.. I see what you mean now. I have removed the call to `getLeaderEpoch`.